doc/regexp.rdoc

```diff
@@ -1,827 +1,1236 @@
-<tt>#{...}</tt> construct.
+- \Method Regexp.new.
 
-    place = "東京都"
-    /#{place}/.match("Go to 東京都")
-        #=> #<MatchData "東京都">
+== \Method <tt>match</tt>
 
-== Character Classes
+Each of the methods Regexp#match, String#match, and Symbol#match
+returns a MatchData object if a match was found, +nil+ otherwise;
+each also sets {global variables}[rdoc-ref:regexp.rdoc@Global+Variables]:
 
-A <i>character class</i> is delimited with square brackets (<tt>[</tt>,
-<tt>]</tt>) and lists characters that may appear at that point in the
-match. <tt>/[ab]/</tt> means _a_ or _b_, as opposed to <tt>/ab/</tt> which
-means _a_ followed by _b_.
+  'food'.match(/foo/) # => #<MatchData "foo">
+  'food'.match(/bar/) # => nil
 
-    /W[aeiou]rd/.match("Word") #=> #<MatchData "Word">
+== Operator <tt>=~</tt>
 
-Within a character class the hyphen (<tt>-</tt>) is a metacharacter
-denoting an inclusive range of characters. <tt>[abcd]</tt> is equivalent
-to <tt>[a-d]</tt>. A range can be followed by another range, so
-<tt>[abcdwxyz]</tt> is equivalent to <tt>[a-dw-z]</tt>. The order in which
-ranges or individual characters appear inside a character class is
-irrelevant.
+Each of the operators Regexp#=~, String#=~, and Symbol#=~
+returns an integer offset if a match was found, +nil+ otherwise;
+each also sets {global variables}[rdoc-ref:regexp.rdoc@Global+Variables]:
 
-    /[0-9a-f]/.match('9f') #=> #<MatchData "9">
-    /[9f]/.match('9f')     #=> #<MatchData "9">
+  /bar/ =~ 'foo bar' # => 4
+  'foo bar' =~ /bar/ # => 4
+  /baz/ =~ 'foo bar' # => nil
 
-If the first character of a character class is a caret (<tt>^</tt>) the
-class is inverted: it matches any character _except_ those named.
+== \Method <tt>match?</tt>
 
-    /[^a-eg-z]/.match('f') #=> #<MatchData "f">
+Each of the methods Regexp#match?, String#match?, and Symbol#match?
+returns +true+ if a match was found, +false+ otherwise;
+none sets {global variables}[rdoc-ref:regexp.rdoc@Global+Variables]:
 
-A character class may contain another character class. By itself this
-isn't useful because <tt>[a-z[0-9]]</tt> describes the same set as
-<tt>[a-z0-9]</tt>. However, character classes also support the <tt>&&</tt>
-operator which performs set intersection on its arguments. The two can be
-combined as follows:
+  'food'.match?(/foo/) # => true
+  'food'.match?(/bar/) # => false
 
-    /[a-w&&[^c-g]z]/ # ([a-w] AND ([^c-g] OR z))
+== Global Variables
+
+Certain regexp-oriented methods assign values to global variables:
+
+- <tt>#match</tt>: see {Method match}[rdoc-ref:regexp.rdoc@Method+match].
+- <tt>#=~</tt>: see {Operator =~}[rdoc-ref:regexp.rdoc@Operator+-3D~].
+
+The affected global variables are:
+
+- <tt>$~</tt>: Returns a MatchData object, or +nil+.
+- <tt>$&</tt>: Returns the matched string, or +nil+.
+- <tt>$`</tt>: Returns the string to the left of the matched string, or +nil+.
+- <tt>$'</tt>: Returns the string to the right of the matched string, or +nil+.
```

Comment:
```suggestion
- <tt>$&</tt>: Returns the matched part of the string, or +nil+.
- <tt>$`</tt>: Returns the part of the string to the left of the match, or +nil+.
- <tt>$'</tt>: Returns the part of the string to the right of the match, or +nil+.
```

i think this would be less ambiguous, especially for `$&`